main.js

```diff
@@ -0,0 +1,97 @@
+// Full disclosure: everything works fine with just one post. Commenting becomes iffy with more than one post.
+
+
+$('#post-submit').on('click', function () {
+
+  var newUniquePost = '<div class="unique-post"></div>'
+  $('#posts').append(newUniquePost);
+
+  var newCommentSection = '<div class="comment-section"><div class="form-holder"></div></div>'
+    
+  var $postMessage = $('#post-message').val();
+  var $posterName = $('#poster-name').val();
+  var $lastUniquePost = $('.unique-post').last();
+
+  $lastUniquePost.append('<p>' + $postMessage + '</p>');
+  $lastUniquePost.append('<p><strong>Posted by: </strong>' + $posterName + '</p>');
+  $lastUniquePost.append('<p class="remove-post">Remove Post</p>')
+  $lastUniquePost.append('<p class="toggle-comments">Show Comments >></p>')
+  
+    
+  $lastUniquePost.append(newCommentSection);
+
+  $lastUniquePost.append('<hr>');
+  
+  $('#post-form')[0].reset();
+
+  manageComments();
+  toggleCommentSection();
+  removePost();
```

Comment:
You've got some major bugs, but it is all from a simple problem.

Everyone one of these functions is binding a click handler - i.e. when you invoke `toggleCommentSection` line 41 starts listening for clicks on the `toggle-comments` button.

So when you add 1 post, this entire function (`#post-submit`'s callback function) is invoked, and `toggleCommentSection` is invoked once and starts listening.

But then when you add another comment, all of the above happens again. So now the callback function for clicking the `toggle-comments` button will get invoked twice (because we've registered that listener twice). Therefore, the comments are shown, then hidden again.

To solve this, I suggest you take these functions out of here and look into understanding how to use `on` to delegate events. Let me know if you have questions about that and I can point you in the right direction.